Expose version information

mhvk opened the pull request:
This is a port of https://github.com/astropy/astropy/pull/9324/commits/e2e285a3512942141875141b6b30bb5c4ae00002, exposing the ERFA and SOFA information. Since we now have our own package, it seemed most logical to expose these inside a new `version` module - and to have that I now use `setuptools_scm` (cc @astrofrog, just to be sure I use it correctly). That solves another thing on the list of #23...

@eteq - I only expose `erfa_version` and `sofa_version`, not the erfa major, minor and micro, since I felt those were not all that useful. I also removed the version functions from `core`, since having a new package would seem reason enough to clean that up (will have to deprecate them in `astropy/_erfa`, I guess).

Since there was a missing link in `README.rst` in building the docs, I took the opportunity to mention how to install using system libraries. Another one off the list in #23...

avalentino commented:
> Looks good, but setuptools_scm should also be added to the pyproject.toml file 

I agree. Looking at the setuptools_scm docs it should be something like::
`"setuptools>=42", "setuptools_scm[toml]>=3.4"`

avalentino commented on setup.cfg:
```diff
@@ -25,6 +24,7 @@ classifiers =
 packages = find:
 requires = numpy
 zip_safe = False
+setup_requires = setuptools_scm
```
> if you don't plan on running setup.py commands directly, but also no harm in leaving it there in case you do want to do e.g. python setup.py --version for instance.

I think that having the possibility to run `python setup.py --version` is very handy for developers,
so i would keep the `setup_requires` entry in in `setup.cfg` (no need to add also numpy and jinja IMHO).

avalentino commented on setup.py:
```diff
@@ -65,4 +65,36 @@ def get_extensions():
+    version = '{version}'
+else:
+    del get_version
+""".lstrip()
```
If you want you could also move the `VERSION_TEMPLATE` contents to a `erfa/version.py.templ` file, bit also as it is now is OK for me.

mhvk replied:
I just copied this method over from astropy. My own sense is to stay close to that since most of us are used to it...

avalentino replied:
OK, makes sense

avalentino commented:
If in #7 we decide to go with the versioning scheme with the postfix pyerfa version number (1.7.0, 1.7.01, ...) it could be useful to customize the setuptools_scm behaviour accordingly:

```python

def guess_next_dev(version):
    from setuptools_scm.version import guess_next_version
    if version.exact:
        return version.format_with("{tag}")
    else:
        version_string = str(version.tag)
        if 'dev' in version_string or len(version_string.split('.')) > 3:
            return guess_next_version(version.tag)
        else:
            guessed = version_string.partition("+")[0] + '.1'
        return version.format_with("{guessed}.dev{distance}", guessed=guessed)

setuptools.setup(
    use_scm_version={
        ...
        'version_scheme': guess_next_dev,},
    ...)
```

mhvk commented:
@avalentino, @astrofrog - OK, I updated the `pyproject.toml` and for good measure also added oldest and dev dependencies test to travis.

@avalentino - thanks also for the nice piece to get the version number! I'd like to leave that for a follow-up PR, though, just to give @eteq a chance to pipe in on the version numbering options in #7.
